rooms.js

```diff
@@ -163,6 +165,110 @@ var Room = (function () {
 		}
 		return true;
 	};
+	//mute handling
+	Room.prototype.runMuteTimer = function () {
+		if (this.muteTimer || this.mutedUsers.length === 0) return;
+
+		var timeUntilExpire = this.mutedUsers[0].time - Date.now();
+		if (timeUntilExpire <= 0) {
+			this.unmute(this.mutedUsers[0].userid, true);
+			//runMuteTimer() is called again in unmute() so this function instance should be closed
+			return;
+		}
+		var self = this;
+		this.muteTimer = setTimeout(function () {
+			self.muteTimer = null;
+			self.runMuteTimer();
+		}, timeUntilExpire);
+	};
+	Room.prototype.isMuted = function (user) {
+		if (!user) return;
+		if (this.mutedUsers) {
+			for (var i = 0; i < this.mutedUsers.length; i++) {
+				if (user.userid === this.mutedUsers[i].userid ||
+					user.guestNum === this.mutedUsers[i].guestNum ||
+					(user.autoconfirmed && user.autoconfirmed === this.mutedUsers[i].autoconfirmed)) {
+					return this.mutedUsers[i].userid;
+				}
+			}
+		}
+	};
+	Room.prototype.getMuteTime = function (user) {
+		var userid = this.isMuted(user);
+		if (!userid) return;
+		for (var i = 0; i < this.mutedUsers.length; i++) {
+			if (userid === this.mutedUsers[i].userid) {
+				return this.mutedUsers[i].time - Date.now();
+			}
+		}
+	};
+	Room.prototype.mute = function (user, setTime) {
+		var userid = user.userid;
+
+		if (!setTime) setTime = 7 * 60000; // default time: 7 minutes
+		if (setTime < 1) setTime = 1; // mostly to prevent bugs
```

Comment:
I don't think we actually need this check anymore. `runMuteTimer()` should already handle this.